Project/Assets/Prairie/Framework/Script/Interaction/Tumble.cs

```diff
@@ -7,31 +7,47 @@ public class Tumble : PromptInteraction
 	/// <summary>
 	/// Allows user to rotate object.
 	/// </summary>
-	private bool pickedUp = false;
+	private bool pickedUp;
+	private Quaternion oldRotation;
+	private Vector3 oldPosition;
+	public float distance = 1.5f;
+	public float speed = 10;
+	private Ray hit;
 
 	// When the user interacts with object, they invoke the ability to 
 	// tumble the object with the I, J, K and L keys. Interacting
 	// with the object again revokes this ability.
 
+	void Start()
+	{
+		pickedUp = false;
+		oldRotation = Quaternion.identity;
```

Comment:
This sets the rotation back to a rotation of 0 in all directions. Often meshes are rotated in the world, and that rotation is lost after finishing tumble in this case. Frequently there are even models that are sideways by default, and unity auto rotates them (this is the case with both Austin and my models). We need to get the actual rotation of the models here, otherwise we get behaviour like [This](https://gfycat.com/DeadlyNarrowHake) (these are all models found in Austin's project)